File: CodingInterview/CodingInterview/Strings/06 Shortest_Word.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace CodingInterview
{
    class _06_Shortest_Word
    {
        public static void Main06_Shortest_Word(string words) {

            //Solution #1
            //var shortest = words.Split(" ").ToArray().Select(w => w.Length).Min();

            //Solution #2 - Iterative
            string strShortest = string.Empty;
            foreach (string w in words.Split(" ")) {
                if (string.IsNullOrEmpty(strShortest))
                    strShortest = w;
                if (w.Length < strShortest.Length) {
                    strShortest = w;
                }
            }

            Console.WriteLine(strShortest.Length);
        }
    }
}

File: CodingInterview/CodingInterview/Strings/06 Shortest_Word.js
/*
SHORTEST WORD

Simple, given a string of words, return the length of the shortest word(s).
String will never be empty and you do not need to account for different data types.
*/

function findShort(s) {
        let warr = s.split(' ');
        let shorter = '';

        warr.forEach(e => {
            if (shorter === '') {
                shorter = e;
            } else if (shorter.length > e.length) {
                shorter = e;
            }
        });

        return shorter.length;
}



//OTHER SOLUTIONS(BP)
function findShort(s) {
    return Math.min(...s.split(" ").map(s => s.length));
}

/*
 TEST CASES
Test.describe("Example tests", _ => {
    Test.assertEquals(findShort("bitcoin take over the world maybe who knows perhaps"), 3);
    Test.assertEquals(findShort("turns out random test cases are easier than writing out basic ones"), 3);
});
 */